api/feedbooks.py

```diff
@@ -0,0 +1,457 @@
+
+    REALLY_IMPORT_KEY = 'really_import'
+    LANGUAGE_KEY = 'language'
+    REPLACEMENT_CSS_KEY = 'replacement_css'
+
+    NAME = ExternalIntegration.FEEDBOOKS
+    DESCRIPTION = _("Import open-access books from FeedBooks.")
+    SETTINGS = [
+        {
+            "key": REALLY_IMPORT_KEY,
+            "type": "select",
+            "label": _("Really?"),
+            "description": _("Most libraries are better off importing free Feedbooks titles via an OPDS Import integration from NYPL's open-access content server or DPLA's Open Bookshelf. This setting makes sure you didn't create this collection by accident and really want to import directly from Feedbooks."),
+            "options": [
+                { "key": "false", "label": _("Don't actually import directly from Feedbooks.") },
+                { "key": "true", "label": _("I know what I'm doing; import directly from Feedbooks.") },
+          ],
+          "default": "false"
+        },
+        {
+            "key": LANGUAGE_KEY,
+            "label": _("Import books in this language"),
+            "description": _("Feedbooks offers separate feeds for different languages. Each one can be made into a separate collection."),
+            "type": "select",
+            "options": [
+                { "key": "en", "label": _("English") },
+                { "key": "es", "label": _("Spanish") },
+                { "key": "fr", "label": _("French") },
+                { "key": "it", "label": _("Italian") },
+                { "key": "de", "label": _("German") },
+            ],
+            "default": "en",
+        },
+        {
+            "key" : REPLACEMENT_CSS_KEY,
+            "label": _("Replacement stylesheet"),
+            "description": _("This will replace the Feedbooks stylesheet with an alternate stylesheet in mirrored copies of the Feedbooks titles. The default value is an accessibility-focused stylesheet produced by the DAISY consortium."),
+            "default": "http://www.daisy.org/z3986/2005/dtbook.2005.basic.css",
+        },
+
+    ]
+
+    BASE_OPDS_URL = u'http://www.feedbooks.com/books/recent.atom?lang=%(language)s'
+
+    THIRTY_DAYS = datetime.timedelta(days=30)
+
+    def __init__(self, _db, collection, *args, **kwargs):
+        kwargs['content_modifier'] = self.replace_css
+        kwargs['data_source_name'] = DataSource.FEEDBOOKS
+
+        integration = collection.external_integration
+
+        really_import = integration.setting(self.REALLY_IMPORT_KEY).bool_value
+        if not really_import:
+            raise Exception("Refusing to instantiate a Feedbooks importer because it's configured to not actually do an import.")
+
+        self.language = integration.setting(self.LANGUAGE_KEY).value
+
+        super(FeedbooksOPDSImporter, self).__init__(_db, collection, **kwargs)
+
+        self.new_css = None
+        new_css_url = integration.setting(self.REPLACEMENT_CSS_KEY).value
+        if new_css_url and self.http_get:
```

Comment:
The content code doesn't have this error checking -- it assumes the GET of the hard-coded CSS document succeeds. I added it because the URL to the document is now configurable and someone might get it wrong.